File: backend/app/schemas/compliance.py
from datetime import date, datetime
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel, Field

from ..models import KYCStatus


class KYCVerifyRequest(BaseModel):
    notes: Optional[str] = None


class KYCRejectRequest(BaseModel):
    # CLAUDE.md §10: on rejection the ball goes back to the customer with the
    # reason — it is never optional here, unlike on verify.
    reason: str = Field(min_length=1)


class ComplianceProfileResponse(BaseModel):
    id: int
    user_id: int
    kyc_status: KYCStatus
    date_of_birth: date
    national_id_number: str
    phone_number: str
    residential_address: str
    employment_status: str
    monthly_income: Decimal
    occupation: str
    has_id_document_back: bool
    has_selfie: bool
    created_at: datetime
    reviewed_by: Optional[int]
    reviewed_at: Optional[datetime]
    review_notes: Optional[str]
    customer_full_name: str
    customer_email: str

    # CLAUDE.md §27 (M11): extended registration fields — optional since a
    # pre-M11 or self-signup profile may not have filled all of these in.
    customer_number: Optional[str] = None
    first_name: Optional[str] = None
    middle_name: Optional[str] = None
    last_name: Optional[str] = None
    id_type: Optional[str] = None
    gender: Optional[str] = None
    nationality: Optional[str] = None
    marital_status: Optional[str] = None
    dependants_count: Optional[int] = None
    phone_number_alt: Optional[str] = None
    next_of_kin_name: Optional[str] = None
    next_of_kin_relationship: Optional[str] = None
    next_of_kin_phone: Optional[str] = None

File: frontend/src/schemas/compliance.ts
import { z } from "zod";

// Mirrors backend app/schemas/compliance.py
export interface ComplianceProfileResponse {
  id: number;
  user_id: number;
  kyc_status: "pending" | "verified" | "rejected";
  date_of_birth: string;
  national_id_number: string;
  phone_number: string;
  residential_address: string;
  employment_status: string;
  monthly_income: string;
  occupation: string;
  has_id_document_back: boolean;
  has_selfie: boolean;
  created_at: string;
  reviewed_by: number | null;
  reviewed_at: string | null;
  review_notes: string | null;
  customer_full_name: string;
  customer_email: string;

  // CLAUDE.md §27 (M11) — optional: a pre-M11 or self-signup profile may not
  // have filled all of these in.
  customer_number: string | null;
  first_name: string | null;
  middle_name: string | null;
  last_name: string | null;
  id_type: string | null;
  gender: string | null;
  nationality: string | null;
  marital_status: string | null;
  dependants_count: number | null;
  phone_number_alt: string | null;
  next_of_kin_name: string | null;
  next_of_kin_relationship: string | null;
  next_of_kin_phone: string | null;
}

/** Shared across every KYC-status display (QueuePage, KycPage,
 * CustomerDetailPage, CustomerListPage) — was 4 byte-identical copies. */
export function kycStatusTone(status: "pending" | "verified" | "rejected"): "success" | "danger" | "warning" {
  if (status === "verified") return "success";
  if (status === "rejected") return "danger";
  return "warning";
}

export const kycRejectSchema = z.object({
  reason: z.string().min(1, "A reason is required"),
});
export type KYCRejectInput = z.infer<typeof kycRejectSchema>;
